static/js/components/subject_page/block.tsx

```diff
@@ -115,33 +116,64 @@ const rankingTileLatestDataAvailableFooter = intl.formatMessage({
 });
 
 /**
- * Helper for determining if we should show the "Snap to date with highest
- * coverage" checkbox.
+ * Helper for determining if we should snap the charts in this block to the
+ * best available coverage
  *
  * Only show the checkbox if:
  * (1) No date is set in the chart config columns (meaning date is "LATEST")
  * (2) Chart types are map and/or ranking
  * @returns boolean
  */
-function shouldShowSnapToLatestData(
+function eligibleForSnapToHighestCoverage(
   columns: ColumnConfig[],
   statVarProvider: StatVarProvider
 ): boolean {
-  const statVarKeys = _.flatten(
-    _.flatten(columns.map((c) => c.tiles.map((tile) => tile.statVarKey)))
-  );
+  const tiles = _.flatten(_.flatten(columns.map((c) => c.tiles)));
+  const statVarKeys = _.flatten(tiles.map((tile) => tile.statVarKey));
   const tileTypes = _.flatten(
     _.flatten(columns.map((c) => c.tiles.map((tile) => tile.type)))
   );
   const statVarSpecs = statVarProvider.getSpecList(statVarKeys);
 
-  const showSnapToLatestData =
+  const isEligibleForSnapToHighestCoverage =
     !_.find<StatVarSpec>(statVarSpecs, (statVarSpec) => !!statVarSpec.date) &&
     !_.find(
       tileTypes,
       (tileType) => tileType !== "MAP" && tileType !== "RANKING"
     );
-  return showSnapToLatestData;
+  return isEligibleForSnapToHighestCoverage;
+}
+
+/**
+ * Helper for determining if we should enable the "Snap to highest coverage"
+ * checkbox.
+ *
+ * Only enable the checkbox if the observations returned from highest coverage
+ * are different from those returned by the latest observation date
+ * (when date=LATEST)
+ * @returns boolean
+ */
+async function shouldEnableSnapToHighestCoverage(
+  placeDcid: string,
+  enclosedPlaceType: string,
+  columns: ColumnConfig[],
+  statVarProvider: StatVarProvider
+): Promise<boolean> {
+  // Check if highest coverage & latest date observations are the same
+  const tiles = _.flatten(_.flatten(columns.map((c) => c.tiles)));
+  const statVarKeys = _.flatten(tiles.map((tile) => tile.statVarKey));
+  const statVarSpecs = statVarProvider.getSpecList(statVarKeys);
```

Comment:
nit: could have a helper for getting all the stat var specs for a list of columns since this code is repeated in isEligibleForSnapToHighestCoverage

Reply:
Good callout with the redundancy. In one of these cases, i'm also extracting the "tile types" as well as stat vars from the columns, and that intermediate step makes it less useful to generalize this to a function (i could do a helper that converts columns to a list of tiles, but that's already a one-liner).

im leaning keeping it as is for now- wdyt?